Python/jit.c

```diff
@@ -113,6 +114,21 @@ mark_executable(unsigned char *memory, size_t size)
 
 // JIT compiler stuff: /////////////////////////////////////////////////////////
 
+#define SYMBOL_MASK_WORDS 4
+
+typedef uint32_t SymbolMask[SYMBOL_MASK_WORDS];
+
+typedef struct {
+    unsigned char *mem;
+    SymbolMask mask;
+    size_t size;
+} TrampolineState;
+
+typedef struct {
+    TrampolineState trampolines;
+    uintptr_t instruction_starts[UOP_MAX_TRACE_LENGTH];
+} CompileState;
```

Comment:
Same here. I like the new compile state a lot, great idea!

```suggestion
typedef struct {
    unsigned char *mem;
    symbol_mask mask;
    size_t size;
} trampoline_state;

typedef struct {
    trampoline_state trampolines;
    uintptr_t instruction_starts[UOP_MAX_TRACE_LENGTH];
} jit_state;
```